webapp/src/app.tsx

```diff
@@ -350,7 +350,7 @@ class ScriptSearch extends data.Component<ISettingsProps, ScriptSearchState> {
                             onClick={() => chgHeader(scr) }
                             />
                     ) }
-                    {galleries.map(scr => <codecard.CodeCardView
+                    {pxt.options.light ? undefined : galleries.map(scr => <codecard.CodeCardView
```

Comment:
You are still paying the price of fetching the galleries form the web in "fetchGalleries". You should shortcut there.
Projects refactoring is long overdue...